scipy/stats/_hypotests.py

```diff
@@ -144,6 +144,219 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    r"""
+    Calculates the poisson means test, the "E-test", for the mean difference of
+    two samples that follow a Poisson distribution from descriptive statistics.
+
+    This is a two-sided test. The null hypothesis is that two independent
+    samples have identical average (expected) values.
+
+    Let :math:`X_{11},...,X_{1n_1}` and :math:`X_{21},...,X_{2n_2}` be
+    independent samples from distributions :math:`Poisson(\lambda_1)` and
+    :math:`Poisson(\lambda_2)`. It is well known that :math:`X_1`
+    and :math:`X_2` are independent:
+
+    .. math:: X_1 = \sum_{i=1}^{n_1} X_{1i} \sim Poisson(n_1\lambda_1)
+
+    .. math:: X_2 = \sum_{i=1}^{n_2} X_{2i} \sim Poisson(n_2\lambda_2)
+
+    Let `count1` and `count2` be the observed values of :math:`X_1` and
+    :math:`X_2`, respectively. The null hypothesis and alternative
+    hypothesis under comparison are
+
+    .. math::
+       H_0: \lambda_1 = \lambda_2 + \mathtt{diff} \quad vs. \quad
+       H_a: \lambda_1 \ne \lambda_2 + \mathtt{diff}
+
+    for ``alternative=two-sided``, where :math:`\mathtt{diff} \ge 0`.
+
+    Parameters
+    ----------
+    k1 : int
+        Sample value of interest from sample 1.
+    n1: float
+        Sample size from sample 1.
+    k2 : int
+        Sample value of interest from sample 2.
+    n2 : float
+        Sample size from sample 2.
+    diff : float, optional
+        The difference of mean between two samples under the null hypothesis
+        (default is 0).
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided': :math:`\lambda_1 \ne \lambda_2 + \mathtt{diff}`
+          * 'less': :math:`\lambda_1 \le \lambda_2 + \mathtt{diff}`
+          * 'greater': :math:`\lambda_1 \ge \lambda_2 + \mathtt{diff}`
+
+    Returns
+    -------
+    statistic : float
+        The test statistic calculated from observed samples.
+    pvalue : float
+        The associated p-value based on the estimated p-value of the
+        standardized difference.
+
+    Notes
+    -----
+    A benefit of the E-test is that it maintains its power even
+    with smaller sample sizes which can reduce sampling costs [1]_. It has
+    been evaluated and determined to be more powerful than the comparable
+    C-test, sometimes referred to as the poisson exact test.
+
+    References
+    ----------
+    .. [1]  Krishnamoorthy, K., & Thomson, J. (2004). A more powerful test for
+       comparing two Poisson means. Journal of Statistical Planning and
+       Inference, 119(1), 23-35.
+
+    .. [2]  Przyborowski, J., & Wilenski, H. (1940). Homogeneity of results in
+       testing samples from Poisson series: With an application to testing
+       clover seed for dodder. Biometrika, 31(3/4), 313-323.
+
+    Examples
+    --------
+
+    Suppose that a gardener wishes to test the number of dodder seeds, a weed,
```

Comment:
```suggestion
    Suppose that a gardener wishes to test the number of dodder (weed) seeds,
```
It is unclear what "a weed" refers to. "weed" typically refers to a grown plant, which is related to but distinct from the seed it grows from, so it seems unusual to call a seed a weed. It could be called "a weed seed", but this is more concise.